Bump jenkins-test-harness to 2.56

slonopotamus opened the pull request:
This PR is expected to reveal bogus Windows tests that do not shutdown things properly. See jenkinsci/jenkins-test-harness#166

### Proposed changelog entries

None

slonopotamus committed:
Bump jenkins-test-harness to 2.56

slonopotamus commented:
Err... Windows CI is still not working?

oleg-nenashev commented:
@slonopotamus Yes, it is still disabled

oleg-nenashev commented:
https://github.com/jenkinsci/jenkins/pull/4155 FTR

slonopotamus committed:
Update hamcrest to use 2.x packaging

See jenkinsci/jenkins-test-harness#161

slonopotamus commented:
Seems like Linux CI is also broken now.

slonopotamus commented:
Do not merge this until Windows CI is working again because this PR is expected to break tests that worked otherwise.

oleg-nenashev commented on pom.xml:
```diff
@@ -91,6 +91,7 @@ THE SOFTWARE.
     <patch.tracker.serverId>jenkins-jira</patch.tracker.serverId>
 
+    <hamcrest.version>2.1</hamcrest.version>
     <matrix-auth.version>2.3</matrix-auth.version>
```
Use dependencyManagement instead?

slonopotamus replied:
jenkins-test-harness [doesn't do that| https://github.com/jenkinsci/jenkins-test-harness/blob/master/pom.xml#L110], why do this here differently?

oleg-nenashev replied:
Less code, but no strong opinion

timja commented:
Windows CI is re-enabled, re-triggering

slonopotamus commented:
Nice. As I expected, some things failed. I plan to fix them within this PR.

slonopotamus committed:
Force garbage collection after stopping plugins

Java doesn't have any direct means to release unreferenced classloaders,
so we force GC to do that so it becomes possible to delete plugin jar on Windows.

alecharp commented:
I added the label WIP based on your latest comment @slonopotamus. Thank you for noticing the problem.

oleg-nenashev commented:
Ready to go @slonopotamus ?

oleg-nenashev reviewed (approved):
I approve it, because the issues seem to be fixed

slonopotamus commented:
This PR only _seems_ to be fixed because Windows CI is disabled.

slonopotamus commented:
JTH was already upgraded to 2.57 in #4392
